mlrun/feature_store/retrieval/spark_merger.py

```diff
@@ -57,30 +57,41 @@ def _generate_vector(
             feature_set = feature_set_objects[name]
             feature_sets.append(feature_set)
             column_names = [name for name, alias in columns]
-            target = get_offline_target(feature_set)
-            if not target:
-                raise mlrun.errors.MLRunInvalidArgumentError(
-                    f"Feature set {name} does not have offline targets"
-                )
+
+            if feature_set.spec.passthrough:
+                if not self.spec.source:
```

Comment:
I don't think `self` (`SparkFeatureMerger`) has a `spec`. Did you mean `feature_set.spec`? If so, how come the test didn't fail?

Reply:
it was not tested correctly. fixed.